RedBeanPHP/OODBBean.php

```diff
@@ -443,6 +443,28 @@ private function getSharedList( $type, $redbean, $toolbox )
 	private function getOwnList( $type, $redbean )
 	{
 		$type = $this->beau( $type );
+		// We won't autoResolve if an explicit alias is set
+		if ( self::$autoResolve && empty( $this->aliasName ) ) {
+			list( $redbean, , $writer, $toolbox ) = $this->beanHelper->getExtractedToolbox();
+			$keyMaps = &$redbean->getKeyMapsCache();
+			if ( !isset( $keyMaps[$type] ) ) {
+				$keyMaps[$type] = $writer->getKeyMapForType( $type );
+			}
+			$keyMap = $keyMaps[$type];
+			$property = $this->__info['type'];
+			$alreadyResolved = FALSE;
+			foreach ( $keyMap as $from => $key ) {
+				if ( $key['table'] == $property ) {
+					// We will autoResolve if there's only one column that could match the type
```

Comment:
So this is different than my version. This renders the feature useless for N-X relations (student/teacher for example).
Also see test: https://github.com/gabordemooij/redbean/blob/12da1519177564e94448a36671e0dcf8b362e6b0/testing/RedUNIT/Base/Aliasing.php

Reply:
but even in this case I could not make it work?

    $Bob   = R::dispense(array('_type'=>'person', 'name'=>'Bob'));
    $Sarah = R::dispense(array('_type'=>'person', 'name'=>'Sarah'));
    $project = R::dispense('project');
    $project->name    = 'Is the earth flat?';
    $project->teacher = $Bob;
    $project->student = $Sarah;
    $id = R::store($project);
    $project = R::load('project', $id);
    var_dump( $project->teacher->name ); //Bob
    $projects = $project->teacher->ownProjectList; //0???
    $projects = $project->student->ownProjectList; //0???

Reply:
In your example, when you call `$project->teacher->ownProjectList;` how can we be sure you want to retrieve the projects where Bob is the teacher only (and not those where he is the student, or all of his projects even) ?

That's what makes it ambiguous to me. In that specific case, both `teacher_id` and `student_id` can link to a `person`, so instead of assuming anything we don't resolve it (and it throws an exception in frozen mode).

While doing `$projects = $project->teacher->alias('teacher')->ownProjectList;` makes it pretty clear.

Now, if there's only one column that could link to a person, then it would get resolved.